hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/FilebasedSchemaProvider.java

```diff
@@ -79,4 +92,80 @@ public Schema getTargetSchema() {
       return super.getTargetSchema();
     }
   }
+
+  private static Schema readAvroSchemaFromFile(String schemaPath, FileSystem fs, boolean sanitizeSchema, String invalidCharMask) {
+    String schemaStr;
+    try (FSDataInputStream in = fs.open(new Path(schemaPath))) {
+      schemaStr = FileIOUtils.readAsUTFString(in);
+    } catch (IOException ioe) {
+      throw new HoodieIOException(String.format("Error reading schema from file %s", schemaPath), ioe);
+    }
+    return parseAvroSchema(schemaStr, sanitizeSchema, invalidCharMask);
+  }
+
+  /*
+   * We first rely on Avro to parse and then try to rename only for those failed.
+   * This way we can improve our parsing capabilities without breaking existing functionality.
+   * For example, we don't yet support multiple named schemas defined in a file.
+   */
+  private static Schema parseAvroSchema(String schemaStr, boolean sanitizeSchema, String invalidCharMask) {
```

Comment:
lets try to use the same variable name, argument name across the code base. 
for eg, shouldSanitize, sanitizeSchema should be unified.